xwakes/wit/component.py

```diff
@@ -385,3 +510,895 @@ def get_shorthand_type(self) -> str:
+
+        source_exponents, test_exponents, plane = _handle_plane_and_exponents_input(
+                                    kind=kind, exponents=exponents,
+                                    source_exponents=source_exponents,
+                                    test_exponents=test_exponents,
+                                    plane=plane)
+
+        # we set impedance and wake to a dummy callable because they will be
+        # overridden by methods
+        super().__init__(impedance=lambda x: 0, wake=lambda x: 0, plane=plane,
+                         source_exponents=source_exponents,
+                         test_exponents=test_exponents,
+                         name="Classic Thick Wall")
+
+    def impedance(self, f):
+        layer = self.layer
+        material_magnetic_permeability = mu0
+        if layer is not None:
+            self._check_layer(layer) # Checks that there are no unsupported layer properties
+                                # (e.g. permeability different from vacuum)
+            material_resistivity = layer.dc_resistivity
+        else:
+            material_resistivity = self.resistivity
+
+        radius = self.radius
+        plane = self.plane
+        factor = self.factor
+        length = self.length
+        exponents = tuple(self.source_exponents + self.test_exponents)
+
+        if plane == 'z' and exponents == (0, 0, 0, 0):
+            out = factor * length * ((1 + np.sign(f)*1j) * material_resistivity
+                            / (2* np.pi * radius)
+                            / self.delta_skin(f, material_resistivity,
+                                              material_magnetic_permeability))
+        # Transverse dipolar impedance
+        elif ((plane == 'x' and exponents == (1, 0, 0, 0)) or
+                (plane == 'y' and exponents == (0, 1, 0, 0))):
+            out = factor * length * ((1 + np.sign(f)*1j) * material_resistivity
+                            / (np.pi * radius**3 * (2 * np.pi * f * np.sqrt(eps0 * mu0)))
+                            / self.delta_skin(f, material_resistivity,
+                                              material_magnetic_permeability))
+        else:
+            raise ValueError("Resistive wall wake not implemented for "
+                  "component {}{}. Set to zero".format(plane, exponents))
+
+        return out
+
+    def wake(self, t, beta0=None):
+        layer = self.layer
+        if layer is not None:
+            self._check_layer(layer) # Checks that there are no unsupported layer properties
+                                # (e.g. permeability different from vacuum)
+            material_resistivity = layer.dc_resistivity
+        else:
+            material_resistivity = self.resistivity
+        radius = self.radius
+        plane = self.plane
+        factor = self.factor
+        length = self.length
+        exponents = tuple(self.source_exponents + self.test_exponents)
+
+        beta0 = beta0 if beta0 is not None else 1
```

Comment:
unused - we need `beta0`  in `function_vs_t` but not here (`wake` should not have it as argument)

Reply:
Removed!